### ISSUE TITLE
Incorrect position used in call to erfa.dtdb when location=None

### ISSUE THREAD

**Issue by paulray** [astropy/astropy#11132]
https://github.com/astropy/astropy/blob/4da65447b13483dbc9b2aa7f1c4a5d133773be86/astropy/time/core.py#L1973-L1975

When a `Time()` has location=None, it is supposed to do the TT to TDB conversion without the topocentric term, which corresponds to passing the coordinates of the geocenter to to `erfa.dtdb`. However, this bit of code is setting the location to (0,0,0) in *geodetic* coordinates, which is at the equator on the surface of the Earth!  This should be changed to `from_geocentric` instead of `from_geodetic`.  This was the cause of this issue in PINT https://github.com/nanograv/PINT/issues/246

**Comment by paulray** [comment]
Here is the relevant part of the erfa docs:

```
*     The topocentric part of the model is from Moyer (1981) and
*     Murray (1983), with fundamental arguments adapted from
*     Simon et al. 1994.  It is an approximation to the expression
*     ( v / c ) . ( r / c ), where v is the barycentric velocity of
*     the Earth, r is the geocentric position of the observer and
*     c is the speed of light.
*
*     By supplying zeroes for U and V, the topocentric part of the
*     model can be nullified, and the routine will return the Fairhead
*     & Bretagnon result alone.
```

So, supplying location=None should result in passing zeros for U and V, but that is not what happens.

**Comment by mhvk** [comment]
Oops!! That will have been hard to track down... Ouch.  Fix in #11134
